ui-v2/src/hooks/global-concurrency-limits.ts

```diff
@@ -0,0 +1,59 @@
+import type { components } from "@/api/prefect";
+import { getQueryService } from "@/api/service";
+import { queryOptions } from "@tanstack/react-query";
+import invariant from "tiny-invariant";
+
+export type GlobalConcurrencyLimit =
+	components["schemas"]["GlobalConcurrencyLimitResponse"];
+export type GlobalConcurrencyLimitsFilter =
+	components["schemas"]["Body_read_all_concurrency_limits_v2_v2_concurrency_limits_filter_post"];
+
+/**
+ * ```
+ *  🏗️ Variable queries construction 👷
+ *  all   =>   ['global-concurrency-limits'] // key to match ['global-concurrency-limits', ...
+ *  list  =>   ['global-concurrency-limits', 'list'] // key to match ['global-concurrency-limits', 'list', ...
+ *             ['global-concurrency-limits', 'list', { ...filter1 }]
+ *             ['global-concurrency-limits', 'list', { ...filter2 }]
+ *  details => ['global-concurrency-limits', 'details'] // key to match ['global-concurrency-limits', 'details', ...]
+ *             ['global-concurrency-limits', 'details', { ...globalConcurrencyLimit1 }]
+ *             ['global-concurrency-limits', 'details', { ...globalConcurrencyLimit2 }]
+ * ```
+ * */
+const QueryKeyFactor = {
+	all: () => ["global-concurrency-limits"] as const,
+	lists: () => [...QueryKeyFactor.all(), "list"] as const,
+	list: (filter: GlobalConcurrencyLimitsFilter) =>
+		[...QueryKeyFactor.lists(), filter] as const,
+	details: () => [...QueryKeyFactor.all(), "details"] as const,
+	detail: (id_or_name: string) =>
+		[...QueryKeyFactor.details(), id_or_name] as const,
+};
+
+export const buildListGlobalConcurrencyLimitsQuery = (
+	filter: GlobalConcurrencyLimitsFilter,
+) =>
+	queryOptions({
+		queryKey: QueryKeyFactor.list(filter),
+		queryFn: async () => {
+			const res = await getQueryService().POST(
+				"/v2/concurrency_limits/filter",
+				{ body: filter },
+			);
+			invariant(res.data);
```

Comment:
Addint this invariant here so that `return res.data` resolves as a `Array<GlobalConcurrencyLimit>` in TS instead of `undefined | Array<GlobalConcurrencyLimit>`

Reply:
If data comes back undefined, returning an empty array would be better than throwing an error. We should get an error on 4xx or 5xx status codes, so returning an empty array should be fine.

Also, I'd rather not introduce a new dependency for such a narrow use case.